### PR TITLE
UI differential drive

### PR DESCRIPTION
**Pull request by shueja**
The switch between swerve and differential is on the dimensions page of robot config, for now. In differential mode, the user needs to specify half of the trackwidth (temporary).

### CODE REVIEW

File: src-core/src/generation/remote.rs [hunk 1]
```diff
@@ -165,6 +165,7 @@ pub async fn remote_generate_parent(
     // check if the solver has already completed
     match serde_json::from_str::<RemoteProgressUpdate>(&o) {
         Ok(RemoteProgressUpdate::CompleteTraj(traj)) => {
+            println!("{:?}", traj);
```
Comment (oh-yes-0-fps) [inline src-core/src/generation/remote.rs:168]: Use tracing where possible

File: src-core/src/generation/generate.rs [hunk 2]
```diff
@@ -350,17 +355,81 @@ pub fn generate(
...
+                modules: config
+                    .modules
+                    .map(|modu: Module<f64>| modu.translation())
+                    .to_vec(),
+            };
+
+            traj_builder.add_progress_callback(swerve_status_callback);
 
-    Ok(postprocess(&result, path, snapshot, counts_vec))
+            traj_builder.set_drivetrain(&drivetrain);
```
Comment (oh-yes-0-fps) [inline src-core/src/generation/generate.rs:379]: See if you can split up the generate function, ill do this further in a future pr but this function is getting quite monolithic

**Review by oh-yes-0-fps — APPROVED** [review]
All the rust and ts in document looks good to me, everything else is beyond me

**Review by spacey-sooty — APPROVED** [review]

**Comment by spacey-sooty** [comment]
Been running this locally it works great

**Pull request merged**